scan/lib/scan/xcpretty_reporter_options_generator.rb

```diff
@@ -64,6 +74,18 @@ def generate_reporter_options
       return reporter
     end
 
+    def generate_xcpretty_args_options
+      xcargs = []
+      unless @xcpretty_args.nil?
+        valid_args = @xcpretty_args & SUPPORTED_XCARGS_TYPES
+        valid_args.each do |raw_arg|
+          arg = raw_arg.strip
+          xcargs << "--#{arg}"
```

Comment:
Going along with my comment of validating arguments, maybe we shouldn't even auto format with `--`. We could just allow `xcpretty_args` to be a string that just gets appended onto the command. This will allow for any future arguments that accept parameters.

Example:
```rb
xcpretty_args: "--tap --no-utf"
```

Thoughts?

Reply:
Yeh, I can see why that makes sense. I suppose I have two minor concerns with this:
1) that it will be used as a first option in place of `formatter`, or even `output_types` in error, which for obvious reasons have better logic and error handling behind them
2) If invalid args are passed, the output isn't as helpful - you just get a stack trace like below, and scan continues without logging anything to xcpretty:
```
/Users/jackspargo/.rvm/gems/ruby-2.4.1/gems/xcpretty-0.3.0/bin/xcpretty:73:in `block in <top (required)>': invalid option: --invalid (OptionParser::InvalidOption)
```

That being said, I've made these changes in this PR - see what you think. I suppose there is a third option here to simple allow any string to be passed in the array and just remove any `SUPPORTED_XCARGS_TYPES` reference. I'd be happy with either solution though to be honest, though I think I'm leaning towards your suggestion in the comment above (that this PR now matches) 😄